### PR TITLE
Help Text Component

### LINKED ISSUE #204: Frontend: HelpText component
Introduce a new component for the helper text seen in the registration flow.

Update existing pages to use the new component instead of having a custom version (student/mentor/admin registration, selectionpage).

image

### PR DESCRIPTION
**Pull request by AkoAyukngat**
# Description
Made a new component(HelpText) and updated the pages with the component to replace the custom version.

Fixes #204 

# How to test?

Storybook

# Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

File: src/client/components/HelpText/HelpText.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.helptext-component {
```
Comment (tpetrina) [inline src/client/components/HelpText/HelpText.css:1]: This file is not formatted.

File: src/client/components/HelpText/HelpText.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    border: 3px;
+    float: left;
+    padding: 30px 40px;
+  }
+  .HelpText-message {
```
Comment (tpetrina) [inline src/client/components/HelpText/HelpText.css:6]: In line one it is `helptext` and here is `HelpText`. Can you align both to one style?

File: src/client/components/HelpText/HelpText.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  return (
+    <div className="helptext-component">
+      <p className={`HelpText-message ${klassName}`}>{txt}</p>
+    </div>
+  );
+}
+HelpText.propTypes = {
+  text: PropTypes.string,
```
Comment (tpetrina) [inline src/client/components/HelpText/HelpText.js:25]: Switch to `children` instead of `text`.

File: src/client/components/HelpText/HelpText.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  let txt = props.text;
+
+  if (!txt) {
+    txt =
+      'If you already do not have a slack id then please make' +
+      ' ' +
+      'one as it is mandatory to have a slack id to connect with' +
+      ' ' +
+      'mentors and other students.';
+  }
```
Comment (tpetrina) [inline src/client/components/HelpText/HelpText.js:16]: You should not have this hard-coded, the text comes from the outside. You are just providing a style.

File: src/client/components/HelpText/HelpText.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+  return (
+    <div className="helptext-component">
+      <p className={`HelpText-message ${klassName}`}>{txt}</p>
+    </div>
+  );
+}
+HelpText.propTypes = {
+  text: PropTypes.string,
+  klass: PropTypes.string,
```
Comment (tpetrina) [inline src/client/components/HelpText/HelpText.js:26]: Rename to `className`.

File: src/client/components/HelpText/HelpText.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './HelpText.css';
+
+export default function HelpText(props) {
+  const klassName = props.klass || '';
```
Comment (tpetrina) [inline src/client/components/HelpText/HelpText.js:6]: Rename to `className` (both).

File: src/client/components/HelpText/HelpText.js (hunk 5 above)
Comment (tpetrina) [reply src/client/components/HelpText/HelpText.js:26]: Also, address the error.

File: src/client/components/HelpText/HelpText.js (hunk 3 above)
Comment (tpetrina) [reply src/client/components/HelpText/HelpText.js:25]: Also, address the error.

File: src/client/containers/RegistrationPage/AdminRegistrationPage/AdminRegistrationPage.component.js [hunk 7]
```diff
@@ -41,13 +41,8 @@ export const AdminRegistrationPage = () => {
...
       </div>
-      <p className="admin-registration-info-text">
-        If you already do not have a slack id then please make one as it is
-        mandatory to have a slack id to connect with mentors and other students.
-      </p>
-      <div className="admin-registration-footer">
-        <Footer />
-      </div>
+
+      <HelpText klass="admin-registration-info-text" />
```
Comment (tpetrina) [inline src/client/containers/RegistrationPage/AdminRegistrationPage/AdminRegistrationPage.component.js:45]: Keep the text here.

File: src/client/containers/RegistrationPage/MentorRegistrationPage/MentorRegistrationPage.component.js [hunk 8]
```diff
@@ -52,10 +53,7 @@ export const MentorRegistrationPage = () => {
         <Button>Submit</Button>
       </div>
 
-      <p className="mentor-registration-info-text">
-        If you already do not have a slack id then please make one as it is
-        mandatory to have a slack id to connect with mentors and other students.
-      </p>
+      <HelpText klass="mentor-registration-info-text" />
```
Comment (tpetrina) [inline src/client/containers/RegistrationPage/MentorRegistrationPage/MentorRegistrationPage.component.js:56]: Keep the text here.

File: src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js [hunk 9]
```diff
@@ -52,10 +53,7 @@ export const StudentRegistrationPage = () => {
       <div className="student-registration-button">
         <Button>Submit</Button>
       </div>
-      <p className="student-registration-info-text">
-        If you already do not have a slack id then please make one as it is
-        mandatory to have a slack id to connect with mentors and other students.
-      </p>
+      <HelpText klass="student-registration-info-text" />
```
Comment (tpetrina) [inline src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js:56]: Keep the text here.

**Review by tpetrina — CHANGES_REQUESTED** [review]
Good start, but should be a bit better.

File: src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js (hunk 9 above)
Comment (AkoAyukngat) [reply src/client/containers/RegistrationPage/StudentRegistrationPage/StudentRegistrationPage.component.js]: Hey Toni, I want to enquire why you asked me to keep the text here, while the component is intended to replace the custom text version.

File: src/client/containers/RegistrationPage/AdminRegistrationPage/AdminRegistrationPage.component.js [hunk 10]
```diff
@@ -41,13 +41,8 @@ export const AdminRegistrationPage = () => {
       <div className="admin-registration-button">
         <Button>Submit</Button>
       </div>
-      <p className="admin-registration-info-text">
-        If you already do not have a slack id then please make one as it is
-        mandatory to have a slack id to connect with mentors and other students.
-      </p>
-      <div className="admin-registration-footer">
-        <Footer />
-      </div>
```
Comment (1620prabn) [inline src/client/containers/RegistrationPage/AdminRegistrationPage/AdminRegistrationPage.component.js:50]: line number 48 to 50  you also deleted footer form the page  <Footer/>

**Comment by AkoAyukngat** [comment]
Too much conflicts.

**Pull request closed without merging**